include/revng/StackAnalysis/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,645 @@
+  /// are used for determining the maximum number of registers to use for
+  /// primitive values that are bigger than a generic registers but don't
+  /// use vector registers (for example `__int128`).
+  static constexpr bool AllowPassingAggregatesInRegisters = false;
+
+  /// States whether specified ABI allows accepting aggregate object arguments
+  /// (structs, unions, etc) in the designated registers if they provide enough
+  /// space (including padding) and the maximum number of registers allowed per
+  /// aggregate argument.
+  ///
+  /// `0` means that using registers for aggregate objects is disallowed.
+  /// `1` means that passing aggregate values are only allowed if their size
+  /// (including padding) doesn't exceed the size of a single generic register.
+  /// `2` means that passing aggregate values are only allowed if their size
+  /// (including padding) doesn't exceed the size of two generic registers.
+  /// ...
+  /// `-1` means that there are no limitations and all the allowed registers
+  /// could be used for a single aggregate argument.
+  ///
+  /// All the non-conforming aggregate values are passed using the stack.
+  ///
+  /// \note: some special cases still apply. For example C++ abi doesn't allow
+  /// structs and classes that are not 'trivial for the purposes of calls' to
+  /// be passed in such a fashion.
+  /// (see https://itanium-cxx-abi.github.io/cxx-abi)
+  ///
+  /// \note: unaligned aggregate objects are always passed using stack.
+  static constexpr size_t GenericRegistersUsedPerAggregateArgument = 0;
+
+  /// States whether specified ABI allows returning aggregate objects (structs,
+  /// unions, etc) in the designated registers if they provide enough space
+  /// (including padding) and the maximum number of registers allowed per
+  /// aggregate argument.
+  ///
+  /// The rules are the same for `GenericRegistersUsedPerAggregateArgument`,
+  /// see relevant comment for additional information.
+  /// \see GenericRegistersUsedPerAggregateArgument
+  ///
+  /// \note The value must either be `0` if using registers for aggregate
+  /// return values is not allowed, `1` if it's allowed for a single register
+  /// only or equal to the size of `GenericReturnValueRegisters` container.
+  static constexpr size_t GenericRegistersUsedPerAggregateReturnValue = 0;
+
+  /// States who is responsible for cleaning the stack after the function call.
+  /// `true` - callee, `false` - caller.
+  static constexpr bool CalleeIsResponsibleForStackCleanup = false;
+
+  /// States the alignment of the stack in bytes.
+  /// \note: states minimum value for abis supporting multiple different stack
+  /// alignment values.
+  static constexpr size_t StackAlignment = 1;
+
+  /// Stores the list of generic registers allowed to be used for argument
+  /// passing in the order they are used.
+  static constexpr std::array<model::Register::Values, 0>
+    GenericArgumentRegisters = {};
```

Comment:
*Generic Registers* are usually known as *General Purpose Register*. Can we adopt this terminology?

Reply:
`GPR` is another kinda well-known abbreviation, if strictly necessary.